Modules/_functoolsmodule.c

```diff
@@ -41,13 +41,31 @@ get_functools_state(PyObject *module)
 
 /* partial object **********************************************************/
 
+
+typedef struct {
+    PyObject_HEAD
+} placeholderobject;
+
+
+static PyTypeObject placeholder_type = {
+    .ob_base = PyVarObject_HEAD_INIT(NULL, 0)
+    .tp_name = "functools.Placeholder",
+    .tp_doc = PyDoc_STR("placeholder for partial arguments"),
+    .tp_basicsize = sizeof(placeholderobject),
+    .tp_itemsize = 0,
+    .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_DISALLOW_INSTANTIATION,
+    .tp_new = PyType_GenericNew,
+};
```

Comment:
https://github.com/python/cpython/actions/runs/9174174702/job/25224470051?pr=119303#step:16:69

> Non-constant global variables are generally not supported
in the CPython repo.

Reply:
I think it is because it didn't have `Py_TPFLAGS_IMMUTABLETYPE` flag, but can not check because the tool is not working for me localy.

Reply:
I think you can mark this as resolved.

Reply:
Finally! `c-analyzer` is not working for `clang` so could not run it locally.